### ISSUE TITLE
migrate away from `lerna`

### ISSUE THREAD

**Issue by vobu** [ui5-community/ui5-ecosystem-showcase#607]
the always present Damocles sword of technical debt and OSS maintenance/funding issues has also hit `lerna`: https://github.com/lerna/lerna/issues/2703
it is not actively maintained anymore. so we
- need to keep v3 and pin that in package.json
- adopt to a new monorepo management solution 
sigh.

**Comment by uxkjaer** [comment]
I'm happy to take on the challenge to migrate to turbo repo and using changesets for versioning. I'll create a pull request when I get some free time and start working on this. I also use pnpm in other repos. Anyone against adding this instead of yarn. I quite like the non flat node modules structure and the cache and symlinking.

**Comment by vobu** [comment]
+1 on _not_ using `yarn` > 1

**Comment by uxkjaer** [comment]
when I replace yarn with pnpm and run install I get this warning:
 WARN  Issues with peer dependencies found
packages/ui5-app
├─┬ @wdio/cli
│ └─┬ @wdio/config
│   └─┬ @wdio/types
│     └── ✕ missing peer typescript@^4.6.2
└─┬ wdio-chromedriver-service
  └── ✕ missing peer webdriverio@^7.0.0
Peer dependencies that should be installed:
  typescript@^4.6.2   webdriverio@^7.0.0

@vobu should I add them as dependencies to the ui5-app or just leave them?

**Comment by uxkjaer** [comment]
Using changesets I believe we can have the github actions bot to create a new pr for a changeset, when that is merged a new version is created on the subpackages. So all is automatically added to the pr when making changes. See this [repo](https://github.com/azu/monorepo-release-changesets) as an example. With Turborepo and husky we can add a pipeline to make sure all "chore" tasks have been completed before push.

**Comment by vobu** [comment]
> @vobu should I add them as dependencies to the ui5-app or just leave them?

those warnings originate from `wdi5` being used - you can ignore them.

**Comment by uxkjaer** [comment]
check this [PR](https://github.com/ui5-community/ui5-ecosystem-showcase/pull/608) for current status and comment if I take it too far

**Comment by vobu** [comment]
to infinity and beyond, brother 👨‍🚀 🤣

**Comment by petermuessig** [comment]
I am currently thinking about using `beachball`: https://microsoft.github.io/beachball/

**Comment by petermuessig** [comment]
Going away from `yarn` may be a bit more complicated... I even see mixing `yarn` and `turborepo` is possible

**Comment by marianfoo** [comment]
i don't know all the consequences, but it's also possible to move all packages in own repos
Just throw it in as a option

**Comment by petermuessig** [comment]
For sure, but as long as one maintains the projects a monorepo is the better approach to make collective changes. Also testing is much simpler for the show-case to just link the related projects using `yarn` workspaces.

**Comment by petermuessig** [comment]
Interesting reading about the package managers: 
https://blog.logrocket.com/javascript-package-managers-compared/

**Comment by petermuessig** [comment]
I need to understand `pnpm` more in depth - as mentioned in the PR installing deps only once makes me interested!

**Comment by uxkjaer** [comment]
Also updating packages in subpackages can be done at root level, it also picks up packages from the repo before npm, so in the ui5 app it will pick up the simple proxy folder instead of the published package which makes it easier to develop and test. How about I give it a go and do some testing, see how I go. I've had very few issues with changing. And the nested package structure can be disabled if we run into issues.

**Comment by petermuessig** [comment]
Yes, please go ahead. Did you also see my change #609 ? Maybe we can submit this one and upgrade to `pnpm` after that. I would also think then of using `beachball` instead of `lerna` for version and release management?

**Comment by uxkjaer** [comment]
Yup sounds great. I'll merge to main and then pull the changes to my branch. Beachball sounds good, I'll take a look. I found changesets a bit cumbersome for our needs.

**Comment by petermuessig** [comment]
Cool. Interesting in #609 are mostly the project configuration files rather than the project sources. There in general `prettier` changed most of the source files to apply the general formatting now.

**Comment by uxkjaer** [comment]
> Interesting reading about the package managers: https://blog.logrocket.com/javascript-package-managers-compared/

just went through the blog. quite interesting. My take on it would be pnpm for now and then maybe yarn berry later on. I'm familiar with pnpm and have had very little issues with it so far.

**Comment by petermuessig** [comment]
I am fine with `pnpm`

**Comment by uxkjaer** [comment]
just had a little play with beachball. It seems really cool. I would suggest we let the CI handle the publish. I tried to add a husky post-commit to handle the beachball change, however it's not interactive then. Any ideas?

**Comment by petermuessig** [comment]
I'll take a look - I wanted to do so nevertheless to get away from `lerna` also for other projects. After you merged my PR, I can do this as a next step to keep the "semi-"automated flow. With `lerna` I also triggered the version changes locally and this finally triggered a commit which triggers the GH release action.

**Comment by petermuessig** [comment]
`lerna` is progressing: https://github.com/lerna/lerna/issues/3121

We'll keep `lerna` for now but we will switch to `pnpm`!
